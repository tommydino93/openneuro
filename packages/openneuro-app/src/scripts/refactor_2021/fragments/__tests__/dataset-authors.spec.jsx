import React from 'react'
import { shallow } from 'enzyme'
import DatasetAuthors from '../dataset-authors.jsx'

describe('DatasetAuthors component', () => {
  it('renders with common props', () => {
    const wrapper = shallow(<DatasetAuthors authors={['One', 'Two']} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('renders when authors is null', () => {
    const wrapper = shallow(<DatasetAuthors />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders null when authors is not an array', () => {
    // Catch the expected PropTypes error
    // eslint-disable-next-line no-console
    console.error = jest.fn()
    const wrapper = shallow(<DatasetAuthors authors={''} />)
    expect(wrapper.getElement()).toBe(null)
  })
})
