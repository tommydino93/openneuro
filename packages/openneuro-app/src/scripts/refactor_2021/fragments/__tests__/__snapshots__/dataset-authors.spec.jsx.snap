// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetAuthors component renders when authors is null 1`] = `
<h6>
  authored by 
</h6>
`;

exports[`DatasetAuthors component renders with common props 1`] = `
<h6>
  authored by One, Two
</h6>
`;
